ostree/ostree_src.go

```diff
@@ -348,7 +356,62 @@ func (s *ostreeImageSource) GetSignatures(ctx context.Context, instanceDigest *d
 	return signatures, nil
 }
 
-// LayerInfosForCopy() returns updated layer info that should be used when reading, in preference to values in the manifest, if specified.
+// LayerInfosForCopy() returns the list of layer blobs that make up the root filesystem of
+// the image, after they've been decompressed.
 func (s *ostreeImageSource) LayerInfosForCopy() []types.BlobInfo {
-	return nil
+	updatedBlobInfos := []types.BlobInfo{}
+	manifestBlob, manifestType, err := s.GetManifest(nil)
+	if err != nil {
+		return nil
+	}
+
+	var schema manifestSchema
+	if err := json.Unmarshal(manifestBlob, &schema); err != nil {
```

Comment:
Please use the `containers/image/manifest` package (`manifest.FromBlob`), both for manifest types and for manifest parsing.

This doesn't work as expected because `LayersDescriptors` and `FSLayers` list layers in the opposite directions; besides, the `manifestSchema` franked-type containing a subset of both schema1 and schema2, and undocumented, is pretty surprising.

Reply:
missed this comment in the last revision.  Going to address it now.

Reply:
done now